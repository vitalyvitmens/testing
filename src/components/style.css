.input-field {
  display: flex;
  flex-flow: column nowrap;

  width: 300px;
  padding: 24px;
}

.input-field label {
  font-size: 10px;
  color: gray;
}

.blackout {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;

  display: flex;
  align-items: flex-start;
  justify-content: center;

  z-index: 10;
  background-color: rgba(0, 0, 0, 0.5);
}

.notifier-wrapper {
  width: 200px;
  padding: 24px;

  font-size: 12px;
  display: flex;
  justify-content: center;
  margin-top: 20px;

  background-color: #fff;
  box-shadow: 0 2px 6px -4px rgba(0, 0, 0, 0.6);
  border-radius: 16px;

  z-index: 20;
}
