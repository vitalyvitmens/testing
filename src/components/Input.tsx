import { ChangeEvent, useState } from 'react'
import './style.css'

type Props = {
  value: string
  onChange: (value: string) => void
}
export const Input = ({ value, onChange }: Props) => {
  const [labelValue, setLabelValue] = useState('')

  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    const val = event.target.value

    if (val.length > 32) {
      setLabelValue(`Превышена максимальная длина заголовка ${val.length}`)
    } else {
      setLabelValue('')
      onChange(val)
    }
  }

  return (
    <div className="input-field">
      <span>Введите заголовок</span>
      <input type="text" value={value} onChange={handleChange} />
      <label>{labelValue}</label>
    </div>
  )
}
