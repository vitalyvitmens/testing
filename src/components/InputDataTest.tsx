import { ChangeEvent, useState } from 'react'
import './style.css'

type Props = {
  value: string
  onChange: (value: string) => void
}
export const InputDataTest = ({ value, onChange }: Props) => {
  const [labelValue, setLabelValue] = useState('')

  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    const val = event.target.value

    if (val.length > 32) {
      setLabelValue(`Превышена максимальная длина заголовка ${val.length}`)
    } else {
      setLabelValue('')
      onChange(val)
    }
  }

  return (
    <div className="input-field">
      <label htmlFor="input-header-field-id" data-test="input-label">
        Введите заголовок
      </label>
      <input
        id="input-header-field-id"
        // aria-hidden={true}
        type="text"
        value={value}
        onChange={handleChange}
        placeholder={`например, введите "купить молоко"`}
        alt="поле для ввода заголовка задачи"
        title="поле для заголовка"
        data-test="input-field"
      />
      <span data-test="input-hint-text">{labelValue}</span>
    </div>
  )
}
