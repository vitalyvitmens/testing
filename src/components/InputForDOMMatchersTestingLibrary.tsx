import { ChangeEvent, useState } from 'react'
import './style.css'

type Props = {
  defaultValue: string
  onChange: (value: string) => void
}
export const InputForDOMMatchersTestingLibrary = ({
  defaultValue,
  onChange,
}: Props) => {
  const [labelValue, setLabelValue] = useState('')
  const [inputValue, setInputValue] = useState(defaultValue)

  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    const val = event.target.value

    if (val.length > 32) {
      setLabelValue(`Превышена максимальная длина заголовка ${val.length}`)
    } else {
      setLabelValue('')
      setInputValue(val)
      onChange(val)
    }
  }

  return (
    <div className="input-field" data-testid="input-container">
      <label htmlFor="input-header-field-id" data-testid="input-label">
        Введите заголовок
      </label>
      <input
        className="input-field-element input-field-node"
        id="input-header-field-id"
        // aria-hidden={true}
        type="text"
        value={inputValue}
        onChange={handleChange}
        placeholder={`например, введите "купить молоко"`}
        alt="поле для ввода заголовка задачи"
        title="поле для заголовка"
        data-testid="input-field"
        // style={{ display: 'none' }}
        // hidden={true}
        // disabled={true}
        style={{ color: 'red', display: 'flex' }}
      />
      <span data-testid="input-hint-text">{labelValue}</span>
    </div>
  )
}
