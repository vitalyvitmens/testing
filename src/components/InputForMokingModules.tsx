import { ChangeEvent, useState } from 'react'
import './style.css'
import { HEADER_LENGTH_ERROR_MESSAGE, validateHeader } from '../utils/helpers'

type Props = {
  defaultValue: string
  onChange: (value: string) => void
}
export const InputForMokingModules = ({ defaultValue, onChange }: Props) => {
  const [labelValue, setLabelValue] = useState('')
  const [inputValue, setInputValue] = useState(defaultValue)

  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    const val = event.target.value

    if (validateHeader(val)) {
      setLabelValue('')
      setInputValue(val)
      onChange(val)
    } else {
      setLabelValue(HEADER_LENGTH_ERROR_MESSAGE)
    }
  }

  return (
    <div className="input-field" data-testid="input-container">
      <label htmlFor="input-header-field-id" data-testid="input-label">
        Заголовок задачи
      </label>
      <input
        id="input-header-field-id"
        type="text"
        className="input-field-element input-field-node"
        value={inputValue}
        onChange={handleChange}
        placeholder={`например, введите "купить молоко"`}
        alt="поле для ввода заголовка задачи"
        title="поле для заголовка"
        data-testid="input-field"
        style={{ display: 'flex' }}
      />
      <span data-testid="input-hint-text">{labelValue}</span>
    </div>
  )
}
