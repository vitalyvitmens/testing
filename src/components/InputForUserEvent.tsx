import { ChangeEvent, useState } from 'react'
import './style.css'

type Props = {
  defaultValue: string
  onChange: (value: string) => void
}
export const InputForUserEvent = ({ defaultValue, onChange }: Props) => {
  const [labelValue, setLabelValue] = useState('')
  const [inputValue, setInputValue] = useState(defaultValue)

  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    const val = event.target.value

    if (val.length > 32) {
      setLabelValue(`Превышена максимальная длина заголовка ${val.length}`)
    } else {
      setLabelValue('')
      setInputValue(val)
      onChange(val)
    }
  }

  return (
    <div className="input-field">
      <label htmlFor="input-header-field-id" data-testid="input-label">
        Введите заголовок
      </label>
      <input
        id="input-header-field-id"
        // aria-hidden={true}
        type="text"
        value={inputValue}
        onChange={handleChange}
        placeholder={`например, введите "купить молоко"`}
        alt="поле для ввода заголовка задачи"
        title="поле для заголовка"
        data-testid="input-field"
      />
      <span data-testid="input-hint-text">{labelValue}</span>
    </div>
  )
}
