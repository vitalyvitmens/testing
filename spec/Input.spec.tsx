import { render, screen, prettyDOM } from '@testing-library/react'
import { Input } from '../src/components/Input'

describe('Поле ввода', () => {
  it.todo('Ограничение на ввод более 32 символов')

  it('Поле доступно для ввода', () => {
    const fn = jest.fn()
    const span = document.createElement('span')

    // const { container } = render(<Input value="Greetings!" onChange={fn} />, {
    //   container: document.body.appendChild(span),
    // })
    // console.log(prettyDOM(container))

    // const { baseElement } = render(<Input value="Greetings!" onChange={fn} />, {
    //   container: document.body.appendChild(span),
    // })
    // console.log(prettyDOM(baseElement))

    // const { rerender } = render(<Input value="Greetings!" onChange={fn} />)
    // screen.debug()
    // rerender(<Input value="Привет!" onChange={fn} />)
    // screen.debug()

    // const { unmount } = render(<Input value="Greetings!" onChange={fn} />)
    // screen.debug()
    // unmount()
    // screen.debug()

    const { asFragment } = render(<Input value="Greetings!" onChange={fn} />)
    const firstRender = asFragment()
    expect(firstRender).toMatchSnapshot()

    // const { getByRole } = render(<Input value="Greetings!" onChange={fn} />)
    // getByRole('text')

    const {} = render(<Input value="Greetings!" onChange={fn} />, {
      container: document.body.appendChild(span),
      hydrate: false,
      legacyRoot: false,
    })
    // screen.debug()
  })
})

// npm test -- Input.spec.tsx --watch
