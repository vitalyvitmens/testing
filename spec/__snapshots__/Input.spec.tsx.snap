// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Поле ввода Поле доступно для ввода 1`] = `
<DocumentFragment>
  <div
    class="input-field"
  >
    <span>
      Введите заголовок
    </span>
    <input
      type="text"
      value="Greetings!"
    />
    <label />
  </div>
</DocumentFragment>
`;
